### PR TITLE
Add the ability to get response headers as a `Map<String, List<String>>`

### PR DESCRIPTION
**Pull request by brianquinlan**
Adds a extension to `BaseResponse` like:

```dart
extension HeadersWithFieldLists on BaseResponse {
  /// The HTTP headers returned by the server.
  ///
  /// The header names are converted to lowercase and stored with their
  /// associated header values.
  ///
  /// Cookies can be parsed using the dart:io `Cookie` class:
  ///
  /// ```dart
  /// import "dart:io";
  /// import "package:http/http.dart";
  ///
  /// void main() async {
  /// final response = await Client().get(Uri.https('example.com', '/'));
  /// final cookies = [
  ///   for (var value i
  ///       in response.headersFieldValueList['set-cookie'] ?? <String>[])
  ///     Cookie.fromSetCookieValue(value)
  /// ];
  Map<String, List<String>> get headersFieldValueList ...
}
```

Fixes https://github.com/dart-lang/http/issues/24

---

- [X] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: pkgs/http/lib/src/base_response.dart [hunk 1]
```diff
@@ -68,3 +70,68 @@ abstract class BaseResponse {
     }
   }
 }
+
+// "token" as defined in RFC 2616, 2.2
```
Comment (natebosch) [inline pkgs/http/lib/src/base_response.dart:74]: [nit] Use `///` and make the comments on private members docs.

File: pkgs/http/lib/src/base_response.dart [hunk 2]
```diff
@@ -68,3 +70,68 @@ abstract class BaseResponse {
...
+  /// import "package:http/http.dart";
+  ///
+  /// void main() async {
+  /// final response = await Client().get(Uri.https('example.com', '/'));
+  /// final cookies = [
+  ///   for (var value i
+  ///       in response.headersFieldValueList['set-cookie'] ?? <String>[])
+  ///     Cookie.fromSetCookieValue(value)
+  /// ];
+  Map<String, List<String>> get headersFieldValueList {
```
Comment (natebosch) [inline pkgs/http/lib/src/base_response.dart:122]: This name is pretty long - can we consider any shorter ones? Maybe `headerSplitValues`?

**Review by natebosch — APPROVED** [review]

File: pkgs/http/lib/src/base_response.dart (hunk 1 above)
Comment (brianquinlan) [reply pkgs/http/lib/src/base_response.dart]: Done

File: pkgs/http/lib/src/base_response.dart (hunk 2 above)
Comment (brianquinlan) [reply pkgs/http/lib/src/base_response.dart]: Done

**Pull request merged**